### DISCUSSION TITLE
Switch to tab N for a particular search engine

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [gdh1995/vimium-c#1053] [Q&A] [answered]
Let's say I have a pinned tab for Google searches, how can I switch to it automatically and perform the search in the pinned tab when using shortcut from vomnibar (let's say g)?

**Comment by gdh1995** [comment]
You may use the `Marks.activateCreate` command to create a global mark by typing `Shift + <letter>`, and then use `Marks.activate` and a same character key to jump to a remembered tab.

If you want to omit the `Shift` key, you may use `map xxx Marks.activateCreate swap` to make lower-case letter key mean a global mark, instead of a local (per-host) one.

If you do want to use Vomnibar, then a nearest way is to type `:pinned` after `Vomnibar.activate` or `Vomnibar.activateTabs`, and Vimium C will show your pinned tabs

**Reply by limaceous-bushwhacker** [reply]
Hey, thanks for the answer. Didn't know we could search for pinned tabs as well. Is it possible to create shortcuts for particular sites within Vimium so we can quick switch to them?
I see your bit about marks but that'd require me to mark sites manually, no? Are global marks permanent? Can I define them in a config file?

**Reply by gdh1995** [reply, marked as answer]
Um, both local and global marks are permanent, except those created in incognito windows.

You can not define Vimium C's "`marks`" using a text editor, because there're quite a few internal details in every created mark.

But here's a similar way:
- if you just want to "**switch to an existing tab or create a new one with a specific URL**`
- you may use:
```
map xxx openUrl url="https://www.bing.com/" reuse="reuse"
# # or just 
# map xxx focusOrLaunch url="https://www.bing.com/"
# and, if you want to avoid reusing a tab in another window:
map yyy openUrl url="https://www.bing.com/" reuse="reuse-in-cur-wnd"
```

**Reply by limaceous-bushwhacker** [reply]
Is there any way to check a list of my marks?

EDIT: While I have your attention, how do I exit Vomnibar on just clicking outside of it? Current behavior just unfocuses it, so to exit Vomnibar I have to refocus it using my mouse and then close it with \<ESC\>

EDIT2: Is it possible to auto-accept if there's only 1 choice available in Vomibar?

**Reply by gdh1995** [reply]
1. see https://github.com/gdh1995/vimium-c/wiki/Show-all-marks
2. You may enable "Vomnibar auto exiting on clicking outside" by adding a parameter named `exitOnClick`, like:
```
map o Vomnibar.activate exitOnClick
```

3. If Vomnibar is unfocused, then you may use `<f2>` to refocus it
4. No, because I think `auto-accept` is not a good feature
- a. If you just want to select a suggestion without pressing `Enter`
- b. then you may: when the keyboard caret is at the end of your query, press 3 spaces
- c. then Vomnibar will open a currently selected suggestion

**Reply by limaceous-bushwhacker** [reply]
Thank you for being so helpful, mate 👏
